STYLEGUIDE.md

```diff
@@ -0,0 +1,133 @@
+
+# Typedefs styleguide
+
+This styleguide aims to answer questions relevants to the cosmetics related to 
+writing Idris programs.
+
+Since idris is a relatively young language, there is no general convention 
+about its style. This document should serve as reference for current and future 
+users of the typedefs codebase.
+
+## Line length: 120
+
+The maximum line length is 120 characters. In today's wide screen monitors 
+there is no reason to limit ourselves to 80 characters, yet, having a maximum
+limit helps avoiding degenerate situations.
+
+## Indent style: 2 spaces
+
+Idris makes uses of some whitespace sensitive construction (typically the `do`
+notation), and also inherits from a culture of ascii-art code from Agda and
+Haskell. In order to allow this level of visual customisation and ensure
+consistency across editors, spaces are used.
+
+The language does not benefit from more than 2 space of indentation in most
+cases. Thought special constructions lend themselve to more spaces for 
+indentation.
+
+### Indentation: where clauses
+
+``` idris
+fn : String
+fn = g
+  where g : String
+        g = "hello"
+```
+
+Where clauses are aligned with 6 spaces
+
+### Indentation: Let declarations
+
+```idris
+fn : String
+fn = let greeting = "hello" in
```

Comment:
What about multiclause let's, i.e., 
```let foo = 1
    bar = 2 
  in
```

Reply:
I had this in mind

```
let foo = 1
    bar = 2 in
    expr
```

What do you think?

Reply:
If you want to align `expr` with the bound variables `foo` and `bar`, why not move the `in` to the next line?
```idris
let foo = 1
    bar = 2
in  expr
```

Reply:
I like @fredrikNordvallForsberg proposal, or otherwise

```
let
  foo = 1
  bar = 2
in
  expr
```

but it starts to be quite long

Reply:
@fredrikNordvallForsberg "If you want to align expr with the bound variables foo and bar, why not move the in to the next line?" 

My argument against it would be that in no other parts of the language or styleguide do we encourage using two spaces between two syntactic terms.

@marcosh I like your proposition better but I agree that it's getting very long. 

If we agree on that indentation

```idris
fn = let a = b in 
         expr
```

(That is, `expr` should be indented with 4 spaces) Then I think it makes sense to stay consistent with the 4 spaces rules and multiple lines indented like so:

```idris
fn = let a = b
         c = d in
         expr
```

Reply:
I've updated the `let` paragraph with 2 spaces instead of 4. Tell me what you think